### PR TITLE
support available path manager

### PR DESCRIPTION
**Pull request by madhanraj**
This commit list down available mptcp path manager

### CODE REVIEW

**Comment by cpaasch** [comment]
Overall looks good! Thanks!

Can you maybe fix the two style-issues and merge all commits into one (it's not worth to split this in three separate commits).

**Comment by PhSchmitt** [comment]
As written on the mailing list, I have two comments:
- it seems we are talking of the path manager. Why do you use MPTCP_SCHED_NAME_MAX (that refers to the scheduler) as maxlen?
- If it doesn't take long, could you please also list all av. schedulers?

**Comment by madhanraj** [comment]
Updated as per comments

File: net/mptcp/mptcp_ctrl.c [hunk 1]
```diff
@@ -92,6 +92,22 @@ static int proc_mptcp_path_manager(struct ctl_table *ctl, int write,
 	return ret;
 }
 
+static int proc_mptcp_available_path_manager(ctl_table *ctl, int write,
+				   void __user *buffer, size_t *lenp,
+				   loff_t *ppos)
+{
+	char val[MPTCP_PM_NAME_MAX];
```
Comment (cpaasch) [inline net/mptcp/mptcp_ctrl.c]: val is too small here, as maxlen is set to MPTCP_PM_BUF_MAX. You will overflow buffers here.

**Pull request closed without merging**